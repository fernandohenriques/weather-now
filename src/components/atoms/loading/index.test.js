import React from 'react';
import { shallow, mount } from 'enzyme';
import Loading from './index';

describe('<Loading />', () => {
  it('Snapshot testing', () => {
    const wrapper = shallow(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Snapshot testing with fullscreen mode', () => {
    const wrapper = shallow(<Loading fullscreen />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should always get the fullscreen prop', () => {
    const wrapper = mount(<Loading fullscreen />);

    expect(wrapper.props().fullscreen).toBe(true);
    wrapper.setProps({ fullscreen: false });
    expect(wrapper.props().fullscreen).toBe(false);
  });

  it('Should fullscreen prop is false if not passed', () => {
    const wrapper = mount(<Loading />);

    expect(wrapper.props().fullscreen).toBeFalsy();
  });
});
