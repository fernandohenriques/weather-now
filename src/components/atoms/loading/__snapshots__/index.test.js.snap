// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Snapshot testing 1`] = `
<div
  className="loading"
>
  <div />
</div>
`;

exports[`<Loading /> Snapshot testing with fullscreen mode 1`] = `
<div
  className="loading fullscreen"
>
  <div />
</div>
`;
